src/test/conditions/prefix_sha256_test.cpp

```diff
@@ -0,0 +1,225 @@
+            check (*f, c.get(), vwxyz, vwxyz);
+        }
+
+        { // A PREFIX-SHA256, with the prefix set to 'abc'
+          // that wraps around an ED25519 condition signing
+          // the message 'abc':
+            auto f = load_fulfillment (
+                "cf:1:A2FiYwAEYHahWSBEpuT1ESZbynOmBNkLBSnR32Ar4woZqSV2YNH1rsara"
+                "pEir_D33Llmf_YTE2iUcytueMJvW2cxAeJn_i4rZfpNU9rUeKGtpk1Q_R39t9l"
+                "JINw-GlZKZHscujVgAQ");
+            BEAST_EXPECT (f);
+
+            {
+                auto const f2 = load_fulfillment(makeSlice(to_blob(*f)));
+                BEAST_EXPECT (f2);
+                BEAST_EXPECT (*f == *f2);
+            }
+
+            auto c = load_condition ("cc:1:25:KHqL2K2uisoMhxznwl-6pai-ENDk2x9Wru6Ls63O5Vs:100");
+            BEAST_EXPECT (c);
+
+            {
+                auto c1 = load_condition (to_string(*c));
+                BEAST_EXPECT (c1);
+                BEAST_EXPECT (*c == *c1);
+
+                auto c2 = load_condition (makeSlice(to_blob(*c)));
+                BEAST_EXPECT (c2);
+                BEAST_EXPECT (*c == *c2);
+            }
+
+            // Ensure that it has the correct features set
+            BEAST_EXPECT (f->requires() ==
+                (feature_sha256 | feature_prefix | feature_ed25519));
+
+            // Test manual construction
+            {
+                prefix_sha256_t f2;
+                f2.setPrefix(abc);
+                f2.setSubfulfillment(load_fulfillment (
+                    "cf:4:dqFZIESm5PURJlvKc6YE2QsFKdHfYCvjChmpJXZg0fWuxqtqkSKv8"
+                    "PfcuWZ_9hMTaJRzK254wm9bZzEB4mf-Litl-k1T2tR4oa2mTVD9Hf232Uk"
+                    "g3D4aVkpkexy6NWAB"));
+
+                // Check the subfulfillment directly:
+                auto sc = f2.subcondition();
+
+                check (f2.subfulfillment(), sc, empty, abc);
+                check (f2.subfulfillment(), sc, abc, abc);
+                check (f2.subfulfillment(), sc, abcd, abc);
+                check (f2.subfulfillment(), sc, vwxyz, abc);
+
+                // This may seem counterintuitive, but it's
+                // not: the subfulfillment signed the message
+                // "abc"; our prefix is also "abc" so in order
+                // to verify this condition successfully, the
+                // message must be empty:
+                check (f2, c.get(), empty, empty);
+                check (f2, c.get(), abc, empty);
+                check (f2, c.get(), abcd, empty);
+                check (f2, c.get(), vwxyz, empty);
+            }
+
+            // Like before, the ED25519 condition we contain
+            // signed the message 'abc' which is our prefix
+            // which means that this will only validate with
+            // an empty message:
+            check (*f, c.get(), empty, empty);
+            check (*f, c.get(), abc, empty);
+            check (*f, c.get(), abcd, empty);
+            check (*f, c.get(), vwxyz, empty);
```

Comment:
In addition to the test for a non-empty message fed to a prefix that you already suggested, how about one where subconditions go more than one deep. So a prefix -> prefix -> ed25519?